context/app/routes_cells.py

```diff
@@ -66,6 +67,39 @@ def _first_n_matches(strings, substring, n):
     } for s, offset in zip(first_n, offsets)]
 
 
+def get_cluster_name_and_number(cluster_str):
```

Comment:
Usually underscore functions that are purely internal to the module.

Can you use (preferably) [a dataclass](https://docs.python.org/3/library/dataclasses.html) or (less preferably) a named tuple? Inside a function, where the dependency is really near by, plain tuples are fine, but when the code is more spread out, structure is good.

For simple functions like this, putting in a doctest is easy:

```suggestion
def _get_cluster_name_and_number(cluster_str):
    '''
    >>> n_n = _get_cluster_name_and_number('cluster-name-number')
    >>> n_n.name
    'cluster-name'
    >>> n_n.number
    'number'
    '''
```

Also: I really don't like getting into splitting IDs / assuming semantics: It's too easy for the API maintainer to forget that part of the contract. Can you make sure there's an issue in the API, and reference it in a TODO here?